image/manifest_test.go

```diff
@@ -48,3 +64,60 @@ func TestUnpackLayerDuplicateEntries(t *testing.T) {
 		t.Fatalf("Expected to fail with duplicate entry, got %v", err)
 	}
 }
+func TestUnpackLayerRemovePartialyUnpackedFile(t *testing.T) {
+	tmp1, err := ioutil.TempDir("", "test-layer")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp1)
+	err = os.MkdirAll(filepath.Join(tmp1, "blobs", "sha256"), 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tarfile := filepath.Join(tmp1, "blobs", "sha256", "test.tar")
```

Comment:
On Wed, Sep 07, 2016 at 09:15:42PM -0700, Lei Jitang wrote:
> > +	tarfile := filepath.Join(tmp1, "blobs", "sha256", "test.tar")
> >
> > test.tar doesn't look like a hex hash ;). I'd rather build this
> > layer outside of blobs/sha256 and then rename it into position
> > once we've completed it and can get a content-addressable hash.
>
> `test.tar` will be rename to a hex hash below.

I'd rather avoid the temporary invalidity by placing it somewhere else
first.  Can we make a ./tmp directory for scratch space and use:

  tempDir := filepath.Join(tmp1, "tmp")
  err := os.Mkdir(tempDir, 0777)
	if err != nil {
    t.Fatal(err)
  }
  f, err := ioutil.TempFile(tempDir, "blob-")

And then use f.Name() where you were using tarfile?